src/main/java/com/tabnine/inline/DebounceManager.kt

```diff
@@ -0,0 +1,28 @@
+package com.tabnine.inline
+
+import com.tabnine.capabilities.CapabilitiesService
+import com.tabnine.capabilities.Capability
+import com.tabnine.userSettings.AppSettingsState
+
+object DebounceManager {
+    private val DEBOUNCE_CAPABILITIES = arrayListOf(
+        Capability.DEBOUNCE_VALUE_300,
+        Capability.DEBOUNCE_VALUE_600
+    )
+
+    @JvmStatic
+    fun getDebounceInterval(): Long {
+        if (CapabilitiesService.getInstance().isCapabilityEnabled(Capability.DEBOUNCE_VALUE_300)) {
+            return 300
+        }
+        if (CapabilitiesService.getInstance().isCapabilityEnabled(Capability.DEBOUNCE_VALUE_600)) {
+            return 600
+        }
```

Comment:
i'd extract this logic to a utility file with a static function that returns `Long?`, and then this code can look like this:
```kotlin
return getDebounceMsFromCapabilities() ?? AppSettingsState.instance.debounceTime
```